packages/react-components/src/components/BaseECharts/eChartsUtlis.ts

```diff
@@ -0,0 +1,107 @@
+import { ECElementEvent, EChartsOption, EChartsType, SeriesOption } from 'echarts';
+import { DataStream } from '@iot-app-kit/charts-core';
+import { StreamType } from '../../common/constants';
+import { DataPoint } from '@iot-app-kit/core';
+import {
+  DEFAULT_MARGIN,
+  tooltipColors,
+  tooltipLineColor,
+  tooltipNameHeight,
+  tooltipNameWidth,
+} from './eChartsConstants';
+import { v4 as uuid } from 'uuid';
+
+// specifically converts dataStreams to series for bar, scatter, line charts
+export const dataStreamsToEChartSeries = (dataStreams: DataStream[], chartType: string): SeriesOption[] =>
+  dataStreams
+    .filter(({ streamType }) => streamType !== StreamType.ALARM) // need to filter out alarm streams. not supported in bar, scatter, line
+    .map(
+      (stream) =>
+        ({
+          name: stream.name ?? '',
+          data: stream.data.map((point: DataPoint) => [point.x, point.y]),
+          type: chartType,
+          itemStyle: { color: stream.color } ?? {},
+          lineStyle: { color: stream.color } ?? {},
+        } as SeriesOption)
+    );
+
+export type GraphicInternal = { timestamp: number[] } & {
+  children: Array<{ shape: { x1: number; x2: number; x: number } }>;
+};
+
+export type SeriesInternal = Array<{ data: Array<number[]> }>;
+
+export const updatePositions = (option: EChartsOption, series: SeriesOption[], chart?: EChartsType) => {
+  const graphic = option.graphic as GraphicInternal[];
+  if (graphic?.length === 0) {
+    return [];
+  }
+  const timestamps = new Set();
+
+  (series as SeriesInternal).forEach((stream) => {
+    stream.data.forEach((data) => {
+      timestamps.add(data[0]);
+    });
+  });
+  const timestampsArray = Array.from(timestamps);
+
+  const widthPerTime = ((chart?.getWidth() ?? 0) - Number(DEFAULT_MARGIN) * 2) / (timestamps.size - 1);
+
+  return graphic?.map((g) => {
+    const index = timestampsArray.indexOf(g.timestamp[0]);
+    const newX = widthPerTime * index + Number(DEFAULT_MARGIN);
+
+    g.children[0].shape.x1 = newX;
+    g.children[0].shape.x2 = newX;
+    g.children[1].shape.x = newX - 30;
+    return g;
+  });
+};
+
+export const getToolTipGraphic = (
```

Comment:
hmmmm, this would be part of the on-click event handler, and move would be when the user moves or when the data changes... we could have the same function for both for both initializing and moving .. for now i have kept them separate given move can be simple.... i am not entirely sure about keeping them in the same context..